gpii/node_modules/solutionsRegistry/test/GenericTermsTests.js

```diff
@@ -42,15 +44,105 @@ gpii.tests.universal.solutionsRegistry.genericTerms.validateSchemas = function (
+                combinedDefaults[termKey] = defaultValue;
+            }
+        });
+
+        // Build a crude rollup of all possible transformations for all solutions.
+        var solutionsDefs = fluid.require(that.options.srPath);
+        var combinedCapabilitiesTransform = {};
+        fluid.each(solutionsDefs, function (solutionDef, solutionKey) {
+            var escapedSolutionKey = "http://registry\\.gpii\\.net/applications/" + solutionKey.replace(/\./g, "\\.");
+            fluid.each(solutionDef.settingsHandlers, function (settingsHandlerDef) {
+                if (settingsHandlerDef.capabilitiesTransformations) {
+                    // Allow for at least a shallow mix of intra-application and common terms transforms.
+                    var filteredTransforms = {};
+                    fluid.each(settingsHandlerDef.capabilitiesTransformations, function (singleTransform, transformKey) {
+                        var hasIntraApplicationTransform = gpii.tests.universal.solutionsRegistry.hasIntraApplicationTransform(singleTransform);
+                        if (!hasIntraApplicationTransform) {
+                            filteredTransforms[transformKey] = singleTransform;
+                        }
+                    });
+                    if (Object.keys(filteredTransforms).length) {
+                        combinedCapabilitiesTransform[escapedSolutionKey] = fluid.extend({}, combinedCapabilitiesTransform[escapedSolutionKey], filteredTransforms);
+                    }
+                }
+            });
+        });
+
+        // Transform the defaults into application-specific terms.
+        var transformedDefaults = fluid.model.transformWithRules(combinedDefaults, combinedCapabilitiesTransform);
+
+        // TODO: Need some way to only checks transforms from common terms and not intra-application transforms.
+        // Now validate against each solution individually.
+        var validationErrorCount = 0;
+        var codex = fluid.require(that.options.codexPath);
+
+        fluid.each(codex, function (solutionSchema, solutionKey) {
+            var toValidate = transformedDefaults[solutionKey];
+            if (toValidate) {
+                var validationResults = globalValidator.validate(solutionSchema, toValidate);
+
+                // Log the failure details.
+                if (validationResults.isError || !validationResults.isValid) {
+                    fluid.log("Error validating transformed default(s) for " + solutionKey + ":");
+                    var localisedErrors = gpii.schema.validator.localiseErrors(validationResults.errors, toValidate);
+
+                    fluid.each(localisedErrors, function (singleError) {
+                        validationErrorCount++;
+                        // TODO: Track down bogus data paths like `.properties['BounceKeysInterval']`
+                        var dataPath = (singleError.dataPath && singleError.dataPath.join) ? singleError.dataPath : [];
```

Comment:
typeof(singleError.dataPath) === "string" is more intelligible

Reply:
The validator should return an array of path segments for dataPath, but there's some issue where it can occasionally be a string that I need to troubleshoot and file a bug for.  For now, I need to avoid using non-array values, I improved the check to make that clearer.